packages/core/upload/server/services/upload.js

```diff
@@ -45,6 +45,20 @@ const sendMediaMetrics = data => {
   }
 };
 
+const addTmpWorkingDirectory = async (files, providedTmpWorkingDirectory) => {
```

Comment:
Do you need the `providedTmpWorkingDirectory` argument? This PR doesn't seem using it anywhere.

I'd call this method `createAndAssignTmpWorkingDirectoryToFiles` instead. I think it needs to be clear the methods actually mutates the files too. Wdyt?